main/src/main/scala/org/clulab/polarity/ml/DeepLearningPolarityClassifier.scala

```diff
@@ -0,0 +1,659 @@
+    *
+    * @param trainingPath Training data
+    * @param trainingRatio Ratio of training samples in the dataset
+    */
+  override def fit(trainingPath:String = spreadsheetPath, trainRatio:Float=0.8.toFloat, saveFlag:Boolean=true): Unit = {
+    
+    if (!_isFitted){
+      val (sens_train, labels_train, sens_test, labels_test) = this.readFromSpreadsheet(trainingPath, trainRatio, maskOption)
+      val N_EPOCH = this.N_EPOCH
+      for (epoch <- 1 to N_EPOCH) {
+        logger.info(s"epoch $epoch")
+        this.fitSingleEpoch(sens_train, labels_train)
+        this.testSingleEpoch(sens_test, labels_test)
+        if (maskOption=="tag") {
+          sgd.learningRate=sgd.learningRate*0.6.toFloat
+        }
+        else{
+          sgd.learningRate=sgd.learningRate*0.3.toFloat
+        }
+      }
+      if (saveFlag) {save()}
+      _isFitted=true
+    }
+
+  }
+
+  /**
+    * Returns whether fit has been called before. Mostly for control
+    *
+    * @return True if the parameters of the model have been fit previously
+    */
+  override def isFitted: Boolean = _isFitted
+
+  /**
+    * Gets the predicted polarity for the provided argument
+    *
+    * @param events Ordered sequence of events to get their polarity from
+    * @return Predictions of Polarity subclasses
+    */
+  override def predict(events: Seq[BioEventMention]): Seq[Polarity] = {
+    var predictions = Seq[Polarity]()
+    for (event<-events) {
+      predictions = predictions:+predict(event)
+    }
+    predictions
+  }
+
+  override def predict(event: BioEventMention): Polarity = {
+    if (event matches "ComplexEvent") {
+
+      var lemmas = event.sentenceObj.words.clone()
+
+      val rule = event.label
+      var rulePolarity = 0
+      if (rule.startsWith("Neg")) {
+        rulePolarity = 0
+      } else {
+        rulePolarity = 1
+      }
+
+
+      val lemmas_masked = maskEvent(lemmas, event, maskOption)
+
+      val y_pred:Expression =
+        this.synchronized{
+          ComputationGraph.renew()
+          runInstance(lemmas_masked, rulePolarity)
+        }
+
+//      scala.io.StdIn.readLine()
+
+      if (y_pred.value().toFloat > 0.5) {
+        PositivePolarity
+      }
+      else {
+        NegativePolarity
+      }
+    }
+    else {NeutralPolarity}
+  }
+
+  def maskEvent(lemmas_raw:Array[String], event:BioEventMention,  maskOption:String): Array[String] ={
+    // filter out the edge cases where event has no controller or controlled.
+    var lemmas = lemmas_raw.clone()
+    if (event.arguments.contains("controller") && event.arguments.contains("controlled")){
+      // filter out the edge cases where controller or controlled is vector.
+      //if (event.arguments("controller").isInstanceOf[mutable.ArraySeq[Mention]] && event.arguments("controlled").isInstanceOf[mutable.ArraySeq[Mention]]) {
+      if (event.arguments("controller").isInstanceOf[Seq[Mention]] && event.arguments("controlled").isInstanceOf[Seq[Mention]]) {
```

Comment:
Please replace all instances of the Strings "controller", "controlled", and "theme" with global variables declared in the class DeepLearningPolarityClassifier, e.g., val CONTROLLER = "controller", etc.

Reply:
@zhengzhongliang

Reply:
Done.